integration_test/test_hyperopt.py

```diff
@@ -0,0 +1,113 @@
+from hyperopt import fmin, tpe, hp, STATUS_OK, STATUS_FAIL
+import hyperopt
+from sigopt.hyperopt import SigOptTrials, upload_trials
+import sigopt
+import numpy as np
+import pytest
+
+
+def objective(p, threshold=0.8):
+  x, y = p['x'], p['y']
+  w = np.random.rand()
+  if w <= threshold:
+    return {
+      'loss': x ** 2 + y ** 2,
+      'sum': x + y,
+      'status': STATUS_OK
+    }
+  else:
+    return {
+      'status': STATUS_FAIL
+    }
+
+objective_random = objective
+objective_success = lambda x: objective(x, 1.0)
+objective_fail = lambda x: objective(x, -1.0)
+objectives = {
+  'random': objective_random,
+  'success': objective_success,
+  'fail': objective_fail
+  }
+
+
+class TestHyperopt(object):
+  def run_fmin(self, online=True, upload=True, objective='random', max_evals=3):
+    project = 'hyperopt-integration-test'
+    trials = SigOptTrials(project=project, online=(online and upload))
+    try:
+      best = fmin(objectives[objective],
+                  space={
+                    'x' : hp.uniform('x', -10, 10),
+                    'y': hp.uniform('y', -10, 10)
+                  },
+                  algo=tpe.suggest,
+                  max_evals=max_evals,
+                  trials=trials)
+    except hyperopt.exceptions.AllTrialsFailed:
+      best = None
+    if upload and not online:
+      trials.upload()
+    return trials, best
+
+  def test_upload_trials(self):
+    trials, _ = self.run_fmin(upload=False)
+    tids = upload_trials('hyperopt-integration-test', trials)
+    self._verify_uploaded_trials(trials, tids)
+
+  @pytest.mark.parametrize("online, upload",
+                           [(False, True),
+                            (True, True),
+                            (False, False),
+                            ])
+  @pytest.mark.parametrize("objective",
+                           ['random',
+                            'success',
+                            'fail'])
+  def test_fmin(self, online, upload, objective, max_evals=3):
+    trials, best = self.run_fmin(online, upload, objective, max_evals)
+    self._verify_best_trial(best, trials, objective)
+    self._verify_runs(trials, upload, max_evals)
+
+  def _verify_best_trial(self, best, trials, objective):
+    if objective == 'fail':
+      assert not best
+    if objective == 'success':
+      assert best
```

Comment:
prefer to use `assert best is None` and `assert best is not None` if appropriate, otherwise `0` could cause the test to fail

Reply:
Done.